core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt

```diff
@@ -94,28 +95,36 @@ private fun <C, R> ColumnGroup<C>.replaceRowsIf(
         .asColumnGroup()
         .cast()
 
+public class UpdateException(override val message: String, cause: Throwable? = null) :
+    IllegalStateException(message, cause),
+    DataFrameError
```

Comment:
Better to add this supertype only if you have example of code that triggers this in compiler plugin

Reply:
I think about it as a way to opt-in reporting warnings for exceptions, that we know for sure make sense for user. In theory we could report any exception. DataFrameError will help us to validate them first